### PR TITLE
Revise git commit format best practice

### PR DESCRIPTION
**Pull request by calebwoods**
- We're moving away from merge commits per se, so we don't need to talk
  about that in particular.
- It is still useful to talk about what goes into a good commit message.
- A lot of our recommended best practices are simply what Deliberate Git
  describes, so this change includes a summary and a link to the
  article.
- Also kept most of the existing content about references to Trello
  cards and whatnot, integrating it with the rest of the changes.

### CODE REVIEW

File: git/03.commit-messages.md [hunk 1]
```diff
@@ -55,6 +55,12 @@ Return 404 on permissions API when no access [TR #45] (GH #4)
 If a user cannot access a study at all we want to be able communicate that to API clients, thus we will return a 404.
 ```
 
+Note: when using abbreviate references to a card or PR that will have
+a limited lifespan if the board or repo host is changed.  If there is
+critical information related to the commit, copy it into the commit message body
+and consider using using fully qualified URLs if you want to ensure link will
+work in the future.
```
Comment (kyptin) [inline git/03.commit-messages.md]: @calebwoods I think the second sentence here makes sense, but I'm struggling to understand what you mean by the first one. :-) Can you clean it up a bit?

**Comment by kyptin** [comment]
I think this still needs review by @BenEinwechter and/or @chorn.

**Comment by chorn** [comment]
I think the summary of Deliberate Git is great.

File: git/03.commit-messages.md (hunk 1 above)
Comment (calebwoods) [inline git/03.commit-messages.md]: Reworded

**Pull request merged**